packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,198 @@
+import { Customer, PaymentSessionStatus } from "../models"
```

Comment:
suggestion: Would change `sessionId` in all methods to a more generic variable name such as `paymentId`.  Also, for refund, capture, etc. session id is misleading as we are actually updating the payment.